vic/drivers/cesm/src/cesm_interface_c.c

```diff
@@ -105,25 +105,26 @@ vic_cesm_run(vic_clock *vclock)
 
     // advance the clock
     advance_time();
-    assert_time_insync(vclock, &dmy);
+    assert_time_insync(vclock, &dmy_current);
 
     // read forcing data
     vic_force();
 
     // run vic over the domain
-    vic_image_run();
+    vic_image_run(&dmy_current);
 
     // return fields to coupler
     vic_cesm_put_data();
 
     // if output:
-    if (check_write_flag(current)) {
+    if (check_write_flag(current - 1)) {
```

Comment:
Is that because `current` has already been advanced before the write check happens? If so, include that specifically as a comment so later users do not stumble over the `-1`. Alternatively rewrite this in a way so that `current` does not get updated till after the `write` happens.